ArduCopter/mode_systemid.cpp

```diff
@@ -296,6 +413,26 @@ void ModeSystemId::log_data() const
     // Full rate logging of attitude, rate and pid loops
     copter.Log_Write_Attitude();
     copter.Log_Write_PIDS();
+
+    if (is_poscontrol_axis_type()) {
+        pos_control->write_log();
+        copter.logger.Write_PID(LOG_PIDN_MSG, pos_control->get_vel_xy_pid().get_pid_info_x());
+        copter.logger.Write_PID(LOG_PIDE_MSG, pos_control->get_vel_xy_pid().get_pid_info_y());
+
+    }
+}
+
+bool ModeSystemId::is_poscontrol_axis_type() const
+{
+    bool ret = false;
+
+    if ((AxisType)axis.get() == AxisType::DISTURB_POS_LAT || (AxisType)axis.get() == AxisType::DISTURB_POS_LONG
+         || (AxisType)axis.get() == AxisType::DISTURB_VEL_LAT || (AxisType)axis.get() == AxisType::DISTURB_VEL_LONG
+         || (AxisType)axis.get() == AxisType::INPUT_LOITER_LAT || (AxisType)axis.get() == AxisType::INPUT_LOITER_LONG) {
+        ret = true;
+    }
```

Comment:
This would be neater as a switch statement.

Reply:
Done. Thanks